python/streamsim/commands.py

```diff
@@ -0,0 +1,65 @@
+import argparse
```

Comment:
In the spirit of “the reviewer calls out every random thought that enters their head”... Do we want to use argparse? The middleware folks just went through [a process](https://jira.lsstcorp.org/browse/RFC-684) to permit the use of [Click](https://click.palletsprojects.com/en/7.x/) instead. I have no horse in this race — it may be that argparse is the better option here for a bunch of reasons, but since Click is the new kid on the block it seems worth at least asking the question.

Reply:
I spent a bit of time looking at Click, yeah. There were a few things that led me away from it:
 - I like that argparse is in the standard library. It's going to be supported approximately forever, so it's good to learn how it works.
 - I find it encouraging that large projects like [`aws-cli` use argparse.|https://github.com/aws/aws-cli/blob/develop/awscli/argparser.py#L32]. This suggests it can scale well.
 - I like how explicit argparse is, compared to click. Take a look at the ["simple click program"|https://click.palletsprojects.com/en/7.x/] advertised on the click homepage - how does argument parsing actually occur? It's not clear to me, and that magic bothers me; I've had too much pain debugging implicit APIs.

None of these are strong reasons one way or another for _this_ particular application. I think the question is "which API should spencer learn, so it's in the toolbelt for the next 5 years?" and argparse seems safer, there.